pkg/recipes/engine/engine.go

```diff
@@ -226,24 +226,30 @@ func (e *engine) getDriver(ctx context.Context, recipeMetadata recipes.ResourceM
 	return definition, driver, nil
 }
 
-func (e *engine) getRecipeConfigSecrets(ctx context.Context, driver recipedriver.Driver, configuration *recipes.Configuration, definition *recipes.EnvironmentDefinition) (v20231001preview.SecretStoresClientListSecretsResponse, error) {
-	secrets := v20231001preview.SecretStoresClientListSecretsResponse{}
+func (e *engine) getRecipeConfigSecrets(ctx context.Context, driver recipedriver.Driver, configuration *recipes.Configuration, definition *recipes.EnvironmentDefinition) (secretData map[string]map[string]string, err error) {
 	driverWithSecrets, ok := driver.(recipedriver.DriverWithSecrets)
 	if !ok {
-		return secrets, nil
+		return nil, nil
```

Comment:
this is existing code. Here the driver does not implement DriverWithSecrets interface and we explicitly return nil. I think it is clearer. We could also return secretData which is not yet initialized and it would amount to returning nil but for better clarity sticking with nil.